src/helpers/time.rs

```diff
@@ -0,0 +1,58 @@
+use std::future::Future;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use std::time::Duration;
+
+/// Simple timer that only works in the presence of tokio runtime. Any other runtime will
+/// make it a no-op.
+#[cfg(not(all(test, feature = "shuttle")))]
+#[pin_project::pin_project]
+pub struct Timer {
```

Comment:
I realized that shuttle tests are broken again after I added tokio sleep because it requires Tokio runtime. So I had to add a wrapper to make it work in both environments